presto-main/src/main/java/com/facebook/presto/operator/repartition/AbstractBlockEncodingBuffer.java

```diff
@@ -377,6 +377,9 @@ public void noMoreBatches()
             bufferAllocator.returnArray(nullsBuffer);
             nullsBuffer = null;
         }
+
```

Comment:
I am still trying to wrap my head around this change: Is this eliminating a leak in that references are never being cleaned or is it shortening the time that references are held ? Because I guess that decodedBlock will anyway be reset when appendData is called when the next page needs to be partitioned ? So it seems that we are cleaning up things for a very short amount of time ?

Reply:
These buffers are around 200mb(as discussed in chat). They will be reset when next data is appended to the same buffer(we have 300+ partition buffers). So keeping 300*200MB is too big

Reply:
I see. 300 partition buffers due to the hash partition count ? Can you add more color why it would be so delayed to clear the buffers ? Like addPage should be called very frequently. So we are holding onto memory very briefly less now. Why is that important?

Reply:
Yes but holding 200MB 300 times(for each hash partition) will not scale as it will equal to 60GB. This is aggravated by the fact that retained size of pages are 200MB(they should be much smaller. our page serialized size limit is 4MB) - which is a bug we need to fix as well.

Reply:
Exactly. So that is why I am curious why this fix works: it does not change the fact that 200x300 still, albeit for less duration. The question is how less ??

Btw, I would not index too much on retained size. This memory accounting stuff is always bogus. The big downside of over accounting is that system will be underutilized unnecessarily. Bad but not ooming.